src/prefect/cli/orion.py

```diff
@@ -114,39 +126,76 @@ async def start(
-                    str(host),
-                    "--port",
-                    str(port),
-                ],
-                env=server_env,
-                stream_output=True,
-            )
+    command = [
+        "uvicorn",
+        "--app-dir",
+        str(prefect.__module_path__.parent),
+        "--factory",
+        "prefect.orion.api.server:create_app",
+        "--host",
+        str(host),
+        "--port",
+        str(port),
+        "--access-log",
+    ]
+
+    pid_file = f"{PREFECT_HOME.value()}/orion.pid"
+    if os.path.exists(pid_file):
+        exit_with_error(
+            "There is already an Orion process running in background.\n"
+            "Stop it with command `prefect orion stop`"
         )
 
-        # Explicitly handle the interrupt signal here, as it will allow us to
-        # cleanly stop the Orion uvicorn server. Failing to do that may cause a
-        # large amount of anyio error traces on the terminal, because the
-        # SIGINT is handled by Typer/Click in this process (the parent process)
-        # and will start shutting down subprocesses:
-        # https://github.com/PrefectHQ/orion/issues/2475
+    if detach is True:
+        try:
+            start_process(command, pid_file, env=server_env)
+            exit_with_success(
+                "Orion running in background.\n"
+                f"Check out the dashboard at {base_url}"
+            )
+        except OSError as e:
+            exit_with_error(e)
+    else:
+        async with anyio.create_task_group() as tg:
+            app.console.print(generate_welcome_blurb(base_url, ui_enabled=ui))
+            app.console.print("\n")
+
+            orion_process_id = await tg.start(
+                partial(
+                    run_process,
+                    command=command,
+                    env=server_env,
+                    stream_output=True,
+                )
+            )
+
+            # Explicitly handle the interrupt signal here, as it will allow us to
+            # cleanly stop the Orion uvicorn server. Failing to do that may cause a
+            # large amount of anyio error traces on the terminal, because the
+            # SIGINT is handled by Typer/Click in this process (the parent process)
+            # and will start shutting down subprocesses:
+            # https://github.com/PrefectHQ/orion/issues/2475
+
+            kill_on_interrupt(orion_process_id, "Orion", app.console.print)
 
-        kill_on_interrupt(orion_process_id, "Orion", app.console.print)
+            app.console.print("Orion stopped!")
 
-    app.console.print("Orion stopped!")
+
+@orion_app.command()
+async def stop():
+    pid_file = f"{PREFECT_HOME.value()}/orion.pid"
+
+    if os.path.exists(pid_file):
+        try:
+            app.console.print("Stopping Orion...")
+            stop_process(pid_file)
+            exit_with_success("Orion stopped!")
+        except ValueError as e:
+            exit_with_error(e)
```

Comment:
What kind of value errors are we catching here? :)

Reply:
I see now in `stop_process` as well this indicates an unparsable PID file. I'd note that here as well.